src/server/scripts/Outland/CoilfangReservoir/TheSlavePens/boss_ahune.cpp

```diff
@@ -0,0 +1,879 @@
+class spell_slippery_floor_periodic : public SpellScriptLoader
+{
+public:
+    spell_slippery_floor_periodic() : SpellScriptLoader("spell_slippery_floor_periodic") { }
+
+    class spell_slippery_floor_periodic_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_slippery_floor_periodic_SpellScript);
+
+        void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+        {
+            if (Unit* target = GetHitPlayer())
+                target->CastSpell(target, SPELL_SLIP);
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_slippery_floor_periodic_SpellScript::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_slippery_floor_periodic_SpellScript();
+    }
+};
+
+// 46146 - Ahune Spanky Hands
+class spell_ahune_spanky_hands : public SpellScriptLoader
+{
+public:
+    spell_ahune_spanky_hands() : SpellScriptLoader("spell_ahune_spanky_hands") { }
+
+    class spell_ahune_spanky_hands_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_ahune_spanky_hands_AuraScript);
+
+        void HandleProc(AuraEffect const* /*aurEff*/, ProcEventInfo& eventInfo)
+        {
+            GetTarget()->CastSpell(eventInfo.GetProcTarget(), SPELL_COLD_SLAP, true);
+        }
+
+        void Register() override
+        {
+            OnEffectProc += AuraEffectProcFn(spell_ahune_spanky_hands_AuraScript::HandleProc, EFFECT_0, SPELL_AURA_PROC_TRIGGER_SPELL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_ahune_spanky_hands_AuraScript();
+    }
+};
+
+class spell_ahune_minion_despawner : public SpellScriptLoader
+{
+public:
+    spell_ahune_minion_despawner() : SpellScriptLoader("spell_ahune_minion_despawner") { }
+
+    class spell_ahune_minion_despawner_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_ahune_minion_despawner_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+            if (GetHitCreature())
+                GetHitCreature()->DespawnOrUnsummon();
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_ahune_minion_despawner_SpellScript::HandleScript, EFFECT_0, SPELL_EFFECT_APPLY_AURA);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_ahune_minion_despawner_SpellScript();
+    }
+};
+
+void AddSC_boss_ahune()
+{
+    new boss_ahune();
+    new npc_frozen_core();
+    new go_ahune_ice_stone();
+    new npc_ahune_bunny();
+    new spell_ahune_synch_health();
+    new spell_summoning_rhyme_aura();
+    new spell_summon_ice_spear_delayer();
+    new spell_ice_spear_control_aura();
+    new spell_slippery_floor_periodic();
+    new spell_ahune_spanky_hands();
+    new spell_ahune_minion_despawner();
```

Comment:
The script looks impressive with immaculate code style. :+1: If you by chance have to edit this file for something else, could you add a NewLine at the end of the file?